runtime/frame.go

```diff
@@ -39,20 +39,43 @@ type Frame struct {
 	globals     *Dict `attr:"f_globals"`
 	lineno      int   `attr:"f_lineno"`
 	code        *Code `attr:"f_code"`
+	taken       bool
 }
 
 // NewRootFrame creates a Frame that is the bottom of a new stack.
 func NewRootFrame() *Frame {
-	return newFrame(nil)
+	f := &Frame{Object: Object{typ: FrameType}}
+	f.pushFrame(nil)
+	return f
 }
 
 // newFrame creates a new Frame whose parent frame is back.
 func newFrame(back *Frame) *Frame {
-	f := &Frame{Object: Object{typ: FrameType}}
+	f := back.frameCache
```

Comment:
I'm confused about this access: doesn't hit have to be `back.threadState.frameCache` since threadState isn't the first anonymous field?